api/api.go

```diff
@@ -0,0 +1,174 @@
+// Package api exposes a low-level interface to the MyRadio API.
+//
+// It exposes the Requester interface for types that represent
+// connections to the API, methods for constructing Requesters, and
+// functions and methods for using Requesters to make requests.
+package api
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strings"
+)
+
+// Request represents an API request being built.
+type Request struct {
+	// The endpoint, as a suffix of the API root URL.
+	Endpoint string
+	// The set of mixins to use.
+	Mixins []string
+	// The map of parameters to use.
+	Params map[string][]string
+}
+
+// NewRequest constructs a new request for the given endpoint.
+func NewRequest(endpoint string) *Request {
+	return &Request{
+		Endpoint: endpoint,
+		Mixins:   []string{},
+		Params:   map[string][]string{},
+	}
+}
+
+// NewRequestf constructs a new request for the endpoint constructed by
+// the given format string and parameters.
+func NewRequestf(format string, params ...interface{}) *Request {
+	return NewRequest(fmt.Sprintf(format, params...))
+}
+
+// Response represents the result of an API request.
+type Response struct {
+	raw *json.RawMessage
+	err error
+}
+
+// IsEmpty checks whether the response payload is present, but empty.
+func (r *Response) IsEmpty() bool {
+	if r.err == nil {
+		return false
+	}
+
+	if r.raw == nil {
+		return true
+	}
+
+	// Check for 'empty' JSON payloads.
+	bs, err := r.raw.MarshalJSON()
+	if err != nil {
+		return false
+	}
+
+	if len(bs) != 2 {
+		return false
+	}
+
+	if bs[0] == '[' && bs[1] == ']' {
+		return true
+	}
+
+	if bs[0] == '{' && bs[1] == '}' {
+		return true
```

Comment:
by which i mean, it might be "empty" as far as the schedule is concerned, but it might not be for a more general response. At the very least requires the original comment readding, and possibly the original function restoring